.github/workflows/ci.yaml

```diff
@@ -55,7 +61,7 @@ jobs:
           # Note: We install in three steps in order to satisfy requirements
           # from non default locations first. Installing the PyTorch CPU
           # wheels saves multiple minutes and a lot of bandwidth on runner setup.
-          pip install --no-compile -r pytorch-cpu-requirements.txt
+          pip install --no-compile -r ${{ matrix.pytorch_requirements }}
```

Comment:
This comment about "Installing the PyTorch CPU wheels" is now outdated. The ordering here interacts with this code: https://github.com/iree-org/iree-turbine/blob/26ff1d28d1e7db58778159a99a435fb22758795a/requirements.txt#L18-L22

In the case of AMDGPU, we want that first so the default pytorch install doesn't pull in the default of CUDA (since we want ROCm instead). On CPU, we want the CPU requirements first because the default of CUDA would work but is slow.

---

I also want to keep it clear via docs and ideally enforced in code that:

* Export from PyTorch to MLIR has no GPU requirement - NVIDIA, AMD, or otherwise
* After export, IREE can run on GPU with no PyTorch requirement
* Interop between PyTorch on GPU and Turbine/Wave/IREE on GPU will need a compatible PyTorch install